fuse_core/src/callback_wrapper.cpp

```diff
@@ -112,23 +112,31 @@ void CallbackAdapter::addCallback(const std::shared_ptr<CallbackWrapperBase> & c
 {
   std::lock_guard<std::mutex> lock(queue_mutex_);
   callback_queue_.push_back(callback);
+
+  // NOTE(CH3): Do we need to keep triggering guard conditions if the queue size is > 1?
```

Comment:
I'm pretty sure the answer is yes, and this is a good catch.

 IIRC the waitable interface assumes that the waitable will become ready by something that will wake the wait set. Guard conditions wake the wait set, but they only wake once no matter how many times they're triggered.

I think a good solution would be to trigger the guard condition in `take_data()`. If after taking data the queue is not emtpy, then trigger the guard condition again so the executor will wake for the next callback in the queue.